layers/+spacemacs/spacemacs-layouts/config.el

```diff
@@ -34,6 +34,9 @@
 (defvar spacemacs--layouts-autosave-timer nil
   "Timer for layouts auto-save.")
 
+(defvar spacemacs-persp-unique-shell nil
+  "If true, the default shell for each layout will have the same 'index' as the layout.")
```

Comment:
I would talk more about what it actually buys the user rather than its technical implementation. Something like "If true, each layout will have its own unique shell."